### PR TITLE
[Bug] Reset selected provider status after loading and before sharing

### PR DESCRIPTION
**Pull request by neokore**
Related to  #997 and #998.

Forces clearing selected provider errors and upload response after selecting a new provider and before sharing a map URL.

### CODE REVIEW

File: src/components/side-panel.js [hunk 1]
```diff
@@ -161,7 +163,10 @@ export default function SidePanelFactory(
       this.props.uiStateActions.toggleModal(SAVE_MAP_ID);
     };
 
-    _onClickShareMap = () => this.props.uiStateActions.toggleModal(SHARE_MAP_ID);
+    _onClickShareMap = () => {
+      this.props.providerActions.resetProviderStatus();
```
Comment (macrigiuseppe) [inline src/components/side-panel.js]: why are we resetting the status when we open the share modal dialog?

we should reset the provider status whenever we complete a sharing action,e .g. sharing successful

Comment (neokore) [reply src/components/side-panel.js]: I applied it when opening to ensure that there is no error pending error in the state from any source.
In any case, I think that it must be reset after closing the save or share dialog in order to show the messages to user; but I'm afraid of showing other pending error message that could not be cleared before.

Comment (heshan0131) [reply src/components/side-panel.js]: `resetProviderStatus` is called when Share map modal is closed. it sets 
```
  isProviderLoading: false,
  providerError: null,
  successInfo: {}
```

It's not very pretty but works. If you need to, we should probably call reset status when close `LoadDataModal` `SaveMapMadal` and `ShareMapModal`

Comment (neokore) [reply src/components/side-panel.js]: Ok, I'll apply it this way.

Comment (neokore) [reply src/components/side-panel.js]: I've applied as you advised and it looks way better, thanks! Is it OK for you?

**Pull request merged**